state/protocol/badger/validity_test.go

```diff
@@ -145,3 +146,53 @@ func TestEntityExpirySnapshotValidation(t *testing.T) {
 		require.NoError(t, err)
 	})
 }
+
+func TestValidateVersionBeacon(t *testing.T) {
+	t.Run("no version beacon is ok", func(t *testing.T) {
+		snap := new(mock.Snapshot)
+
+		snap.On("VersionBeacon").Return(nil, nil)
+
+		err := validateVersionBeacon(snap)
+		require.NoError(t, err)
+	})
+	t.Run("height must be below highest block", func(t *testing.T) {
+		snap := new(mock.Snapshot)
+		block := unittest.BlockFixture()
+		block.Header.Height = 12
+
+		vb := &flow.SealedVersionBeacon{
+			SealHeight: uint64(37),
+		}
+
+		snap.On("Head").Return(block.Header, nil)
+		snap.On("VersionBeacon").Return(vb, nil)
+
+		err := validateVersionBeacon(snap)
+		require.Error(t, err)
```

Comment:
If we use sentinels in `validateVersionBeacon`, we can explicitly check that the specific expected error is being returned here

InvalidServiceEventError
```suggestion
		require.ErrorIs(t, err, protocol.InvalidServiceEventError)
```